### PR TITLE
Changes to remove startup packages from being cooked when we export a…

### PR DESCRIPTION
**Pull request by ReneAtAround**
Changes to remove startup packages from being cooked when we export assets or scenes.

I changed the EmptyPackageList function to remove more startup packages when we export assets or scenes. I also made the GetUnsolicitedPackages function virtual so we can better filter out invalid assets that should not be cooked.

### CODE REVIEW

File: Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp
Comment (etiennebeaucheminp-prompto) [inline Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp]: Can you put in the comment the **reason why** we don't want startup packages?

File: Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp
Comment (etiennebeaucheminp-prompto) [inline Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp]: Why does it say `SoftObject`? Isn't that just a name?

File: Engine/Source/Editor/UnrealEd/Classes/CookOnTheSide/CookOnTheFlyServer.h
Comment (etiennebeaucheminp-prompto) [inline Engine/Source/Editor/UnrealEd/Classes/CookOnTheSide/CookOnTheFlyServer.h:373]: I guess there are some changes related to that in our code base?

Comment (ReneAtAround) [reply Engine/Source/Editor/UnrealEd/Classes/CookOnTheSide/CookOnTheFlyServer.h:373]: Yes, but I'm waiting until this is merged before I submit.

File: Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp
Comment (ReneAtAround) [reply Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp]: done.

File: Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp
Comment (ReneAtAround) [reply Engine/Source/Editor/UnrealEd/Private/CookOnTheFlyServer.cpp]: I renamed this. It was copied from another part of the engine.

**Review by etiennebeaucheminp-prompto — APPROVED** [review]

**Pull request merged**
